web/pgadmin/dashboard/static/js/Dashboard.jsx

```diff
@@ -833,6 +839,26 @@ export default function Dashboard({
     );
   };
 
+  const CustomActiveOnlyHeaderLabel =
+    {
+      label: gettext('Active sessions only'),
+    };
+  const CustomActiveOnlyHeader = () => {
+    return (
+      <InputCheckbox
+        label={gettext('Active sessions only')}
+        labelPlacement="end"
+        className={classes.searchInput}
+        onChange={(e) => {
+          e.preventDefault();
+          setActiveOnly(e.target.checked);
+          setRefresh(!refresh);
+        }}
+        value={activeOnly}
+        controlProps={CustomActiveOnlyHeaderLabel}
+      ></InputCheckbox>);
+  };
+    
   return (
```

Comment:
To filter on front end:
```
const finalDashData = useMemo(()=>{
  if(tabVal == 0 && activeOnly) {
    return dashData.filter((r)=>(r.state=='active'));
  }
  return dashData;
}, [activeOnly, tabVal])
```
and pass finalDashData to all the tables.

Reply:
this helped immensely, a little more digging and I had it working well. I ended up adding dashData to the array of monitored objects as I couldn't tell if that would inherently trigger (and my browser seems to not clear cache reliably, so it was going to take a long time to determine it manually).